docs/_spec/05-classes-and-objects.md

```diff
@@ -989,106 +984,119 @@ transparent trait Enum extends Any, Product, Serializable:
    ```scala
    case ´C_1´, ..., ´C_n´
    ```
-   expands to
+   expands to the following simple enum cases
    ```scala
    case ´C_1´; ...; case ´C_n´
    ```
    Any modifiers or annotations on the original case extend to all expanded cases.
-   This result is then further rewritten by either (3 or 4).
+   <p>This result is then further rewritten by either (3 or 4).</p>
 
-3. A singleton case without an extends clause
+3. A simple enum case `´C´` of an unparameterized enum `´E´` without type parameters
    ```scala
    case ´C´
    ```
-   of an unparameterized enum `´E´` expands to the following simple enum case in `´E´`'s companion object:
+  expands to the following value enum case:
    ```scala
-   val ´C´ = $new(n, "C")
+   case ´C´ extends ´E´
    ```
-   Here, `$new` is a private method that creates an instance of ´E´ (see below).
+   This result is then further rewritten with rule (8).
 
-4. A singleton case without an extends clause
+4. A simple enum case  `´C´` of an unparameterized enum `´E´[´\mathit{tps}´]` with type parameters
    ```scala
    case ´C´
    ```
-   of an enum `´E´` with type parameters
+   where `´\mathit{tps}´` are of the following form
    ```scala
    ´\mathit{v}_1´ ´T_1´ >: ´L_1´ <: ´U_1´ ,   ... ,   ´\mathit{v}_n´ ´T_n´ >: ´L_n´ <: ´U_n´      (n > 0)
    ```
-   where each of the variances `´\mathit{v}_i´` is either `'+'` or `'-'`, expands to the following value enum case:
+   and where each of the variances `´\mathit{v}_i´` is either `'+'` or `'-'`, expands to the following value enum case:
    ```scala
    case ´C´ extends ´E´[´B_1´, ..., ´B_n´]
    ```
    where `´B_i´` is `´L_i´` if `´\mathit{v}_i´ = '+'` and `´U_i´` if `´\mathit{v}_i´ = '-'`.
-   This result is then further rewritten with rule (8).
-   **NOTE:** It is not permitted for enums with non-variant type parameters to have singleton cases without an extends clause.
+   <p>This result is then further rewritten with rule (8).</p>
 
-5. A class case without an extends clause
+5. A class enum case with type parameters, but without an extends clause
    ```scala
-   case ´C´ <type-params> <value-params>
+   case ´C´[´\mathit{tps}´](´\mathit{ps}_1\,´)...(´\mathit{ps}_n´)
    ```
-   of an enum `´E´` that does not take type parameters expands to
+   of an unparameterized enum `´E´` without type parameters expands to
    ```scala
-   case ´C´ <type-params> <value-params> extends ´E´
+   case ´C´[´\mathit{tps}´](´\mathit{ps}_1\,´)...(´\mathit{ps}_n´) extends ´E´
    ```
    This result is then further rewritten with rule (9).
 
-6. If `´E´` is an enum with type parameters `´\mathit{tps}´`, a class case with neither type parameters nor an extends clause
+6. A class enum case without type parameters or an extends clause
    ```scala
-   case ´C´ <value-params>
+   case ´C´(´\mathit{ps}_1\,´)...(´\mathit{ps}_n´)
    ```
-   expands to
+   of an unparameterized enum `´E´[´\mathit{tps}´]` with type parameters expands to
```

Comment:
we can work on the exact terminology, but these desugaring rules where the extends clause is inferred requires that the base enum has no term parameters - so I was calling this an "unparameterized enum". Maybe you could suggest a better term than "enum without term parameters, but has type parameters"

Reply:
Ah, OK, that's the same misunderstanding as above. It would be good to find a better name because "parameterized" refers to *type* parameters everywhere else in the spec. But it's not an issue with this PR's changes, so we can defer that.

Reply:
ok then this is to revisit